v2/apiserver/internal/core/events.go

```diff
@@ -532,6 +535,29 @@ func (e *eventsService) GetByWorkerToken(
 	return event, nil
 }
 
+func (e *eventsService) Clone(
+	ctx context.Context,
+	id string,
+) (EventList, error) {
+	events := EventList{}
+
+	if err := e.authorize(ctx, system.RoleReader(), ""); err != nil {
```

Comment:
I think this is the wrong role. One would need to have `core.ProjectUser` (for the event's project) in order to be authorized to create any event at all.

Reply:
Here I decided to remove the authz prior to the `store.Get` call -- and just defer to the authz checks in the subsequent `Create()` call.  I noticed that `UpdateSourceState`, `Cancel`, `Delete` all follow this pattern.  WDYT?

Reply:
Maybe just leave a comment that makes note of that?